docs/content/concepts/io-management/io-managers.mdx

```diff
@@ -376,7 +427,30 @@ If you're working with time window partitions, you can also use the `asset_parti
 
 #### Handling partition mappings <Experimental />
 
-If you're using a <PyObject object="PartitionMapping" />, a single partition of one asset might depend on a range of partitions of an upstream asset. In that case, you can access the entire range using <PyObject object="InputContext" method="asset_partition_key_range" />, <PyObject object="InputContext" method="asset_partition_keys" />, or <PyObject object="InputContext" method="asset_partitions_time_window" />.
+A single partition of one asset might depend on a range of partitions of an upstream asset.
+
+The default IO Manager has support for loading multiple upstream partitions. In this case, the downstream asset should use `Dict[str, ...]` type for the upstream `DagsterType` (otherwise an error will be raised). Here is an example of loading multiple upstream partitions:
```

Comment:
I might have missed this in the original implementation, but I don't think we should raise an error in this case. In general, I don't think behavior should be dependent on getting type annotations right (with the exception being cases where behavior is undefined without a type annotation, like when an I/O manager is trying to determine what type to load an input as and has multiple options).

Reply:
Yeah, we can skip checking the type if the use didn't provide a type annotation. 

However, I think that if the user did provide a type and it's not a Dict[str, ...], we should still raise an error, because it becomes impossible for Dagster to perform type-checking in this case. This is usually done for normal assets (if the types are specified). Is this correct?

Reply:
I modified the code to skip type checking for the `Any` typing_type

Reply:
> However, I think that if the user did provide a type and it's not a Dict[str, ...], we should still raise an error, because it becomes impossible for Dagster to perform type-checking in this case.

That sounds reasonable to me. Will it work if it's a `Mapping`?

Reply:
No, it doesn't. I tried.